planner/core/task.go

```diff
@@ -1483,6 +1572,71 @@ func (p *PhysicalHashAgg) cpuCostDivisor(hasDistinct bool) (float64, float64) {
 	return math.Min(float64(finalCon), float64(partialCon)), float64(finalCon + partialCon)
 }
 
+func (p *PhysicalHashAgg) attach2TaskForMpp(tasks ...task) task {
+	t := tasks[0].copy()
+	mpp, ok := t.(*mppTask)
+	if !ok {
+		return invalidTask
+	}
+	inputRows := mpp.count()
+	switch p.MppRunMode {
+	case Mpp1Phase:
+		/// 1-phase agg: when the partition columns can be satisfied, where the plan does not need to enforce Exchange
+		/// only push down the original agg
+		prop := &property.PhysicalProperty{TaskTp: property.MppTaskType, ExpectedCnt: math.MaxFloat64, PartitionTp: property.HashType, PartitionCols: mpp.hashCols}
+		proj := p.convertAvgForMPP(prop)
+		p.self.SetChildren(mpp.p)
+		mpp.p = p.self
+		if proj != nil {
+			proj.SetChildren(mpp.p)
```

Comment:
we can use attachPlan2Task

Reply:
`attachPlan2Task()` introduces one more copy overhead.

Reply:
and I think cleaner code is more important here

Reply:
updated